### ISSUE TITLE
Deprecate at-evalpoly macro in favor of function

### ISSUE THREAD

**Issue by musm** [JuliaLang/julia#33289]
Deprecate at-evalpoly macro in favor of function (https://github.com/JuliaLang/julia/pull/32753)

**Comment by rapus95** [comment]
Just out of curiosity, why the macros won't be deprecated in the next minor version? Does deprecation count as breaking change? Because if not, people can start rewriting their uses right now. And if breaking deprecations are simply guaranteed to be only removed at the next major version this would be ok with semver as I understand it.

**Comment by stevengj** [comment]
Yes, I would say that a deprecation of previously exported/documented functionality counts as a breaking change.   (This is what used to happen pre-1.0 — you would update to a new version, and previously working code would still work but with hundreds of deprecation warnings streaming by.  People would consider the package partly broken until the deprecation was fixed.)
